### PR TITLE
feat: 🎸 Generators dto's

### PR DESCRIPTION
**Pull request by iYadgar**
created script to generate all generators dto's according to angular's cli schema.json

✅ Closes: 70

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: [CONTRIBUTING.md](CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+const { readdirSync, readFileSync, writeFile } = require('fs');
+const { join } = require('path');
+const PATH =
+  '/Users/idan.yadgar/angular-cli-e-square/angular-cli/packages/schematics/angular';
```
Comment (eladzipper96) [inline script.js:4]: Is this left by mistake?

File: script.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+const { readdirSync, readFileSync, writeFile } = require('fs');
```
Comment (OmerGronich) [inline script.js:1]: Why not make this a TS file?

File: script.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+const { join } = require('path');
+const PATH =
+  '/Users/idan.yadgar/angular-cli-e-square/angular-cli/packages/schematics/angular';
+
+function capitalize(s) {
+  return s[0].toUpperCase() + s.slice(1);
+}
```
Comment (OmerGronich) [inline script.js:8]: export it to a generic place for utils

Comment (OmerGronich) [inline script.js:8]:
```suggestion
function capitalize(s) {
  return s[0].toUpperCase() + s.slice(1).toLowerCase();
}
```

File: script.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  });
+  return `
+${getPropertiesImports(properties, Boolean(requiredProps.length))}
+export class Generate${schema.name}Dto {
+    ${propsDecorators.toString().split(',').join(' ')}
+    }
+    `;
+};
+const pathToSave =
+  '/Users/idan.yadgar/angular-cli-e-square/angular-cli-gui/apps/cli-daemon/src/app/generators/dto/generators';
```
Comment (OmerGronich) [inline script.js:92]: ?

**Review by OmerGronich — CHANGES_REQUESTED** [review]
Cool!

Few questions

When is the script going to run? Are we going to run it periodically? 
And how are we going to make sure that the DTO's it generates are actually correct? We need to write a lot of tests for this script.

**Review by ronnetzer — COMMENTED** [review]
Continuing my question in your HLD, what about "workspace schematics"? (Is it possible with ng cli?), what about 3rd party schematics? how do we plan to "discover" schematics on run-time? as I understand this is a compile-time thing, we run this script before we deploy a new version.
if we do need to do this stuff in run-time, there is no point of generating DTOs, we should parse schema.json and map it to a general DTO, something like `SchematicArgument[]` (maybe there's an interface from angular we can use) when the client requests for the available arguments of some schematic. I guess we can also generate the class in run-time in order to use it's class-validator features when BE receives an execute/generate command

**Pull request closed without merging**